src/utils/workspace.ts

```diff
@@ -66,7 +71,28 @@ export async function showWorkspaceFoldersQuickPick(placeHolderString: string, c
                 return vscode.workspace.getConfiguration(extensionPrefix, f.uri).get(subPathSetting);
             }
             return;
-        }
-
+        },
+        fileExtension
     );
 }
+
+export function getContainingWorkspace(fsPath: string): vscode.WorkspaceFolder | undefined {
+    // tslint:disable-next-line:strict-boolean-expressions
+    const openFolders: vscode.WorkspaceFolder[] = vscode.workspace.workspaceFolders || [];
+    return openFolders.find((f: vscode.WorkspaceFolder): boolean => {
+        return isPathEqual(f.uri.fsPath, fsPath) || isSubpath(f.uri.fsPath, fsPath);
+    });
+}
+
+export async function findFilesByFileExtension(deployPath: string | undefined, fileExtension: string): Promise<IAzureQuickPickItem<string | undefined>[]> {
+    // if there is a deployPath, then only check the deployed folder for the file extension, otherwise use all currently opened workspaces
```

Comment:
This file should have generic workspace utils, so you shouldn't reference a specific command like "deploy". Just do "folderPath" or "fsPath".

Also this returns quick picks? That seems weird to me. I would expect it to return the files.